src/location.cpp

```diff
@@ -22,7 +22,17 @@ bool is_archive(std::string_view s) {
          ends_with(s, ".tar.zst");
 }
 
-static inline bool is_github(std::string_view s) { return starts_with(s, "https://github.com"); }
+constexpr std::string_view github_com = "https://github.com/";
+
+std::string_view normalize(std::string_view loc) {
+   if (is_github_repo(loc)) {
+      return strip_github_url_to_shorthand(loc);
```

Comment:
I suggest to combine it with `is_github_repo` or refactor in other way. Currently this function doesn't check for `github.com` to be at the beginning. That means that its call will break a string if called without `is_github_repo`.
That is error prone. I would either call `is_github_repo` inside `strip_github_url_to_shorthand` or make a generic `strip_host_from_url` function that will be called like that: `strip_host_from_url(github_com, loc)`

Reply:
Combined `normalize` and `strip_github_url_to_shorthand` and this function is called now `strip_github_com`